src/main/fc/core.c

```diff
@@ -517,7 +517,12 @@ void tryArm(void)
             } else {
                 flipOverAfterCrashActive = true;
 #ifdef USE_RUNAWAY_TAKEOFF
-                runawayTakeoffCheckDisabled = false;
+                if (pidConfig()->runaway_takeoff_prevention_crashflip_reset) {
+                    runawayTakeoffCheckDisabled = false;
+                } else {
+                  runawayTakeoffCheckDisabled = true;
+                }
```

Comment:
This logic is incorrect and creates an unexpected safety feature bypass. With this logic if the user has the reset disabled and activates/deactivates crash flip **BEFORE** the first actual flight arming, then runaway takeoff prevention will be unexpectedly disabled. This is not the intent. The logic should simply be as follows with no `else` clause:
```
                 if (pidConfig()->runaway_takeoff_prevention_crashflip_reset) {
                     runawayTakeoffCheckDisabled = false;
                 }
```